### PR TITLE
Export regions in gltf

### PR DESCRIPTION
**Pull request by knervous**
This exports region data with the associated meshes if the flag is set to true. Metadata includes the region types and zoneline information for the consumer.

### CODE REVIEW

File: LanternExtractor/EQ/Wld/Exporters/ActorGltfExporter.cs [hunk 1]
```diff
@@ -172,8 +172,17 @@ private static void ExportZone(WldFileZone wldFileZone, Settings settings, ILogg
             var textureImageFolder = $"{wldFileZone.GetExportFolderForWldType()}Textures/";
             gltfWriter.GenerateGltfMaterials(materialLists, textureImageFolder);
 
+            var bspNodes = wldFileZone.GetFragmentsOfType<BspTree>();
             foreach (var mesh in zoneMeshes)
             {
+                if (settings.ExportZoneMeshGroups)
```
Comment (vermadas) [inline LanternExtractor/EQ/Wld/Exporters/ActorGltfExporter.cs]: Looks like your PR is missing the corresponding settings.txt and Settings.cs changes.

Comment (knervous) [reply LanternExtractor/EQ/Wld/Exporters/ActorGltfExporter.cs]: Was initially piggybacking off the `ExportZoneMeshGroups` setting but this probably deserves its own setting either way--updated the settings here

File: LanternExtractor/EQ/Wld/Exporters/GltfWriter.cs [hunk 2]
```diff
@@ -291,6 +292,37 @@ public void GenerateGltfMaterials(IEnumerable<MaterialList> materialLists, strin
             }
         }
 
+        public void AddRegionData(Mesh mesh, DataTypes.BspNode frag)
+        {
+            var meshBuilder = new MeshBuilder<VertexPositionNormal>(mesh.Name);
+            var materialBuilder = new MaterialBuilder("");
+            var meshHelper = new WldMeshHelper(mesh, _separateTwoFacedTriangles);
+            var polygonCount = mesh.MaterialGroups.Aggregate(0, (acc, val) => acc + val.PolygonCount);
```
Comment (vermadas) [inline LanternExtractor/EQ/Wld/Exporters/GltfWriter.cs]: TIL about LINQ Aggregate method. Though I wonder if it's more readable to do:
`mesh.MaterialGroups.Sum(m => m.PolygonCount)`
Assuming I'm understanding your intent.

Comment (knervous) [reply LanternExtractor/EQ/Wld/Exporters/GltfWriter.cs]: TIL about Sum! That's a nice replacement. Changed

**Pull request merged**